cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/ImpersonatedTwillController.java

```diff
@@ -78,13 +79,22 @@ public ServiceDiscovered discoverService(String serviceName) {
   public ResourceReport getResourceReport() {
     try {
       return impersonator.doAs(programId.getNamespaceId(), new Callable<ResourceReport>() {
+        @Nullable
         @Override
         public ResourceReport call() throws Exception {
           return delegate.getResourceReport();
         }
       });
     } catch (Exception e) {
-      LOG.warn("Failed in getting resource report for {}.", programId, e);
+      // The delegate.getResourceReport() call never throws exception, hence exception caught must due to
```

Comment:
It never throws exception including runtime. The contract is to return null if it can't fetch the report due to whatever reason